### ISSUE TITLE
Bundle from multiple root files or merge paths

### ISSUE THREAD

**Issue by uncaught** [Redocly/redocly-cli#1303]
**Describe the problem to be solved**
Splitting up API schemas into multiple files is somewhat tedious if you have a lot of different routes.

Currently I have to do this pattern with over 100 routes in my index.yml (set as root in .redocly.yaml):

```yml
paths:
  /rest/billings:
    $ref: './Billing/Billings.yml#/paths/~1rest~1billings'
  /rest/billings/{billingId}:
    $ref: './Billing/Billings.yml#/paths/~1rest~1billings~1{billingId}'
  /search/billings:
    $ref: './Billing/Billings.yml#/paths/~1search~1billings'
  /export/billings/{billingId}/json:
    $ref: './Billing/Billings.yml#/paths/~1export~1billings~1{billingId}~1json'
  # ... there are more here

  /rest/customers:
    $ref: './Customer/Customers.yml#/paths/~1rest~1customers'
  /rest/customers/{customerId}:
    $ref: './Customer/Customers.yml#/paths/~1rest~1customers~1{customerId}'
  /search/customers/{ident}/{value}:
    $ref: './Customer/Customers.yml#/paths/~1search~1customers~1{ident}~1{value}'
  /search/customers:
    $ref: './Customer/Customers.yml#/paths/~1search~1customers'
  # ... more and more

  # ... and also more files that are included with their many routes
```

Each referenced file is a qualified openapi schema file, so that the IDE picks up the file's schema for code completion. E.g. Billings.yml:

```yml
openapi: 3.0.0
info:
  title: ''
  version: ''

paths:
  /rest/billings:
    get:
# ...
```

I hope I'm not missing anything, but I searched over and over again for a better solution. If you know one, please tell me.

This pattern becomes unmaintainable especially since you have to escape all the referenced slashes in the path name.

**Describe the solution you'd like**

I would like to do something as simple as this:

```yml
paths:
  allOf:
    - $ref: './Billing/Billings.yml'
    - $ref: './Customer/Customers.yml'      
    # ...
```

In other terms, I don't want to repeat all the paths over and over again. They are already defined in the referenced files.

**Describe alternatives you've considered**

I understand that this is part of the openapi schema and thus probably not valid and misplaced, so how about allowing multiple files in the .redocly.yaml as root? Or additional files there? E.g.:

```yml
apis:
  core@v1:
    root:
      - ./index.yml # for common settings
      - ./Billing/Billings.yml
      - ./Customer/Customers.yml
    rules:
      no-ambiguous-paths: error
    decorators:
      remove-x-internal: on
```

**Additional context**

I tried using the `bundle` command with multiple root files, but it creates multiple outputs, not one bundled output, unfortunately.

**Comment by uncaught** [comment]
> Please let us know if any of this helps?

Hey there, I just had to add another bulk of routes and gave `redocly join` a shot. It looks very promising, indeed!

There is just one problem: If I use `--without-x-tag-groups` it won't bundle my manually configured `x-tagGroups` from my big file.

Furthermore it seems like `join` wants to put all routes from one file into their own `tagGroup`. But in my use case I have all routes from one file in one `tag` and then have multiple files in one `tagGroup`. Maybe this behavior can be configured in the future, like a switch for "file == tag" vs. "file == tagGroup".

For now I helped myself by putting my tagGroups into a separate yaml file and just concat it onto the bundled one coming out of `join`. This works great!
